assets/javascripts/discourse/widgets/discourse-reactions-counter.js.es6

```diff
@@ -104,5 +124,88 @@ export default createWidget("discourse-reactions-counter", {
   _hoverHandler(event) {
     this.callWidgetFunction("cancelCollapse");
     this.callWidgetFunction("toggleStatePanel", event);
+  },
+
+  collapsePanels() {
+    this.cancelCollapse();
+
+    this.state.statePanelExpanded = false;
+    this.state.reactionsPickerExpanded = false;
+
+    const container = document.getElementById(this.buildId(this.attrs));
+    container &&
+      container
+        .querySelectorAll(
+          ".discourse-reactions-state-panel.is-expanded, .discourse-reactions-reactions-picker.is-expanded"
+        )
+        .forEach(popper => popper.classList.remove("is-expanded"));
+
+    this.scheduleRerender();
+  },
+
+  scheduleCollapse() {
+    this._collapseHandler && cancel(this._collapseHandler);
+    this._collapseHandler = later(this, this.collapsePanels, 500);
+  },
+
+  cancelCollapse() {
+    this._collapseHandler && cancel(this._collapseHandler);
+  },
+
+  toggleStatePanel(event) {
+    if (!this.state.statePanelExpanded) {
+      this.expandStatePanel(event);
+    }
+  },
+
+  expandStatePanel() {
+    this.state.reactionsPickerExpanded = false;
+    this.state.statePanelExpanded = true;
+    this.scheduleRerender();
+    this._setupPopper(this.attrs.post.id, "_popperStatePanel", [
+      ".discourse-reactions-state-panel"
+    ]);
+  },
+
+  _setupPopper(postId, popperVariable, selectors) {
```

Comment:
I would prefer popper instead of popperVariable and popperElement instead of popper